lib/storage/src/content_manager/toc.rs

```diff
@@ -308,6 +309,27 @@ impl TableOfContent {
         Ok(true)
     }
 
+    fn on_peer_failure_callback(&self) -> replica_set::OnPeerFailure {
+        let proposal_sender = self.consensus_proposal_sender.clone();
+        Box::new(move |peer_id| {
+            let proposal_sender = proposal_sender.clone();
+            Box::new(async move {
+                proposal_sender
+                    .send(ConsensusOperations::CollectionMeta(
+                        CollectionMetaOperations::SetShardReplicaState(SetShardReplicaState {
+                            // TODO: fill in actual collection name and shard_id
```

Comment:
Can you fix this `todo` as well? I think it is in scope of this PR.

Reply:
Already on it :+1: 

It seems `ShardId` needs to be passed through the callback as well.

Reply:
Yes, shouldn't be a problem I think